tests/parser.cpp

```diff
@@ -3376,6 +3376,275 @@ import foo; begin { }
 )");
 }
 
+TEST(Parser, order_of_operations)
```

Comment:
Just some drive by additional tests after the parser refactoring? Any reason not to include some `if` conditionals without the parens tests?